src/Microsoft.Health.Checkpoints/Storage/StorageCheckpointClient.cs

```diff
@@ -0,0 +1,143 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Health.Checkpoints.Storage
+{
+    public class StorageCheckpointClient : ICheckpointClient
+    {
+        private BlobContainerClient _storageClient;
+        private ILogger<StorageCheckpointClient> _logger;
+        private const string _lastProcessedDateTime = "LastProcessedDateTime";
+        private const string _lastProcessedIdentifier = "LastProcessedIdentifier";
+
+        public StorageCheckpointClient(BlobContainerClient containerClient, ILogger<StorageCheckpointClient> logger)
+        {
+            EnsureArg.IsNotNull(containerClient);
+
+            _storageClient = containerClient;
+            _logger = logger;
+        }
+
+        public virtual async Task<ICheckpoint> GetCheckpointAsync(string partition, string checkpointIdentifier)
+        {
+            EnsureArg.IsNotNullOrEmpty(partition, nameof(partition));
+            EnsureArg.IsNotNullOrEmpty(checkpointIdentifier, nameof(checkpointIdentifier));
+
+            ICheckpoint checkpoint = new Checkpoint();
+
+            try
+            {
+                var resultSegment = _storageClient.GetBlobsAsync(
```

Comment:
nit: use explicit type when the type is not clear from method.

Reply:
updated